modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -327,33 +373,199 @@ object Run extends ScalaCommand[RunOptions] {
               logger,
               allowExecve = allowExecve
             )
-            Right(proc)
+            Right((proc, None))
           }
         }
       case Platform.JVM =>
-        if (showCommand) {
-          val command = Runner.jvmCommand(
-            build.options.javaHome().value.javaCommand,
-            build.options.javaOptions.javaOpts.toSeq.map(_.value.value),
-            build.fullClassPath,
-            mainClass,
-            args,
-            useManifest = build.options.notForBloopOptions.runWithManifest
-          )
-          Left(command)
-        }
-        else {
-          val proc = Runner.runJvm(
-            build.options.javaHome().value.javaCommand,
-            build.options.javaOptions.javaOpts.toSeq.map(_.value.value),
-            build.fullClassPath,
-            mainClass,
-            args,
-            logger,
-            allowExecve = allowExecve,
-            useManifest = build.options.notForBloopOptions.runWithManifest
-          )
-          Right(proc)
+        runMode match {
+          case RunMode.Default =>
+            if (showCommand) {
+              val command = Runner.jvmCommand(
+                build.options.javaHome().value.javaCommand,
+                build.options.javaOptions.javaOpts.toSeq.map(_.value.value),
+                build.fullClassPath,
+                mainClass,
+                args,
+                useManifest = build.options.notForBloopOptions.runWithManifest,
+                scratchDirOpt = scratchDirOpt
+              )
+              Left(command)
+            }
+            else {
+              val proc = Runner.runJvm(
+                build.options.javaHome().value.javaCommand,
+                build.options.javaOptions.javaOpts.toSeq.map(_.value.value),
+                build.fullClassPath,
+                mainClass,
+                args,
+                logger,
+                allowExecve = allowExecve,
+                useManifest = build.options.notForBloopOptions.runWithManifest,
+                scratchDirOpt = scratchDirOpt
+              )
+              Right((proc, None))
+            }
+          case RunMode.SparkSubmit =>
```

Comment:
NIT I wonder if it wouldn't make sense to extract the Spark-related logic to a trait, like `RunSpark` (so that we could keep `Run.scala` at a reasonable line count)

Reply:
Good idea, done.